public/app/features/alerting/unified/components/AlertLabels.tsx

```diff
@@ -1,26 +1,56 @@
 import { GrafanaTheme } from '@grafana/data';
 import { useStyles } from '@grafana/ui';
 import { css } from '@emotion/css';
-import React, { FC } from 'react';
+import React, { useCallback } from 'react';
 import { AlertLabel } from './AlertLabel';
+import { SilenceMatcher } from 'app/plugins/datasource/alertmanager/types';
 
-interface Props {
-  labels: Record<string, string>;
-}
+type LabelsProps = { labels: Record<string, string>; onRemoveLabel?: (index: number) => void; matchers?: never };
+type MatchersProps = { matchers: SilenceMatcher[]; onRemoveLabel?: (index: number) => void; labels?: never };
 
-export const AlertLabels: FC<Props> = ({ labels }) => {
-  const styles = useStyles(getStyles);
+type Props = LabelsProps | MatchersProps;
 
-  // transform to array of key value pairs and filter out "private" labels that start and end with double underscore
-  const pairs = Object.entries(labels).filter(([key]) => !(key.startsWith('__') && key.endsWith('__')));
+export const AlertLabels = ({ labels, matchers, onRemoveLabel }: Props) => {
+  const styles = useStyles(getStyles);
 
-  return (
-    <div className={styles.wrapper}>
-      {pairs.map(([key, value]) => (
-        <AlertLabel key={`${key}-${value}`} labelKey={key} value={value} />
-      ))}
-    </div>
+  const removeLabel = useCallback(
+    (index: number) => {
+      if (!!onRemoveLabel) {
+        onRemoveLabel(index);
+      }
+    },
+    [onRemoveLabel]
   );
+
+  if (labels) {
+    const pairs = Object.entries(labels).filter(([key]) => !(key.startsWith('__') && key.endsWith('__')));
+
+    return (
+      <div className={styles.wrapper}>
+        {pairs.map(([key, value]) => (
+          <AlertLabel key={`${key}-${value}`} labelKey={key} value={value} />
+        ))}
+      </div>
+    );
+  }
+  if (matchers) {
```

Comment:
this looks liek two different components smushed togeher. Imho `<Matchers/>` should be a separate component